[oauth2] Misc fixes and UX improvements

nyalldawson opened the pull request:
Some more minor fixes and UX improvements to the oauth2 authentication provider.

nyalldawson committed:
[oauth2] Use token url if refresh token url is not specified

The refresh token url is marked as optional -- so we fallback
on the token url if user has not specified a specific refresh URL

nyalldawson committed:
Fix some clazy warnings

nyalldawson committed:
[outh2] Don't crash when trying to parse an invalid json configuration file

nyalldawson committed:
[oauth2] Show a list of directories searched for predefined configuration
files in the list tooltip/help text

Immediately gives users more clues about how to use these
files and where they must be placed

nyalldawson commented:
@elpaso look ok to you?

elpaso commented on src/auth/oauth2/qgso2.cpp:
```diff
@@ -69,7 +69,8 @@ void QgsO2::initOAuthConfig()
 
   setTokenUrl( mOAuth2Config->tokenUrl() );
-  setRefreshTokenUrl( mOAuth2Config->refreshTokenUrl() );
+  // refresh token url is marked as optional -- we use the token url if user has not specified a specific refresh URL
```
This doesn't look right to me: refresh URL is optional because the refresh operation is not always supported, but I'm pretty sure that if the refresh endpoint exists, it cannot be the same URL of the token URL.

elpaso commented:
Nice improvements! Just one thing to change.

nyalldawson committed:
Fix failing oauth2 test

The presence of the state parameter which has no QPROPERTY
counterpart in QgsAuthOAuth2Config was causing the
conversion for json to qobject to fail.
